apps/veil/src/pages/tournament/ui/delegator-table/delegator-rewards.tsx

```diff
@@ -21,39 +51,134 @@ export const DelegatorRewards = () => {
             Total Rewards Earned
           </Text>
           <Text small color='text.secondary'>
-            Cumulative from all epochs, voting and LPs rewards
+            Cumulative from all epochs, and voting rewards
           </Text>
         </div>
-
-        {isLoading || !total ? (
-          <div className='w-24 h-10'>
-            <Skeleton />
-          </div>
-        ) : (
-          <div className='flex items-center gap-4 [&_span]:font-mono [&_span]:text-3xl'>
-            <Density sparse>
-              <ValueViewComponent valueView={total} priority='tertiary' />
-            </Density>
-          </div>
-        )}
+        {totalChild}
       </div>
+      {tableChild}
+    </div>
+  );
+};
 
-      <div className='flex flex-col gap-4 mt-4'>
-        <div className='[&_button]:grow'>
-          <SegmentedControl value={tab} onChange={value => setTab(value as typeof tab)}>
-            <SegmentedControl.Item style='filled' value='lp'>
-              LPs Rewards
-            </SegmentedControl.Item>
-            <SegmentedControl.Item style='filled' value='voting'>
-              Voting Rewards
-            </SegmentedControl.Item>
-          </SegmentedControl>
-        </div>
-
-        {tab === 'lp' && <LpRewards />}
+export const DelegatorRewards = ({ address }: { address: Address }) => {
+  const getMetadata = useGetMetadata();
+  const [page, setPage] = useState(BASE_PAGE);
+  const [limit, setLimit] = useState(BASE_LIMIT);
+  const { getTableHeader, sortBy } = useSortableTableHeaders<SortKey>('epoch', 'desc');
 
-        {tab === 'voting' && <VotingRewards />}
+  const onLimitChange = (l: number) => {
+    setLimit(l);
+    setPage(BASE_PAGE);
+  };
+  const { data, isPending, error } = useSpecificDelegatorRewards({
```

Comment:
praise: wow, calling Pindexer without writing an endpoint manually just works!

thoughts (non-blocking, only trying to udnerstand): This API here looked like magic to me, thanks for this new perspective. Previously, I was trying SSR multiple times, and each time it turned out to be a real pain. The magic of this API is that it's actually not the SSR, it's [a server action](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) with only job is to save some lines of code of creating and calling the endpoint.

Under the hood it creates a POST endpoint for the same page route as we have in NextJs. And calling the function sends a request to this endpoint from client-side environment, what a great idea!